packages/uni_app/lib/view/schedule/schedule.dart

```diff
@@ -67,7 +67,34 @@ class SchedulePageViewState extends State<SchedulePageView>
       length: 6,
     );
 
-    final weekDay = widget.currentWeek.start.weekday;
+    var weekDay = widget.currentWeek.start.weekday;
+
+    final lecturesThisWeek = <Lecture>[];
+    widget.currentWeek.weekdays.take(6).forEach((day) {
+      final lectures = lecturesOfDay(widget.lectures, day);
+      lecturesThisWeek.addAll(lectures);
+    });
+
+    if (lecturesThisWeek.isNotEmpty) {
+      final now = DateTime.now();
+      Lecture? nextLecture;
+      Duration? closestDuration;
+
+      for (final lecture in lecturesThisWeek) {
+        if (lecture.endTime.isAfter(now)) {
+          final difference = lecture.endTime.difference(now);
+          if (closestDuration == null || difference < closestDuration) {
+            closestDuration = difference;
+            nextLecture = lecture;
+          }
+        }
+      }
```

Comment:
This can be changed for something like `final nextLecture = lecturesThisWeek.firstWhere((lecture) => lecture.endTime.isAfter(widget.now));`

I am assuming that there are no concurrent lectures and that they are ordered in the list (I think they are)

Reply:
`lecturesThisWeek` isn't in chronological order, it always starts on Monday and ends on Saturday even if the next class is on a Tuesday.

 I made changes to use `where` and `fold` functions instead of this to make the code cleaner.

Reply:
@limwa is this closed?